library/ssl_tls.c

```diff
@@ -2605,6 +2615,14 @@ static int ssl_tls13_session_load(mbedtls_ssl_session *session,
     memcpy(session->resumption_key, p, session->resumption_key_len);
     p += session->resumption_key_len;
 
+#if defined(MBEDTLS_SSL_EARLY_DATA)
+    if (end - p < 4) {
```

Comment:
```suggestion
   MBEDTLS_SSL_CHK_BUF_PTR(p, end, 4);
```

Reply:
It is not in `ssl` context, this macro can not be used.

Reply:
Seems this check is not related with the ssl context, only name it as `ssl_`. 
"#define MBEDTLS_SSL_CHK_BUF_PTR(cur, end, need)                        \
    do {                                                                 \
        if (mbedtls_ssl_chk_buf_ptr((cur), (end), (need)) != 0) \
        {                                                                \
            return MBEDTLS_ERR_SSL_BUFFER_TOO_SMALL;                  \
        }                                                                \
    } while (0)"

Reply:
see `ssl_tls12_session_load` also. These functions can not use the macros. This place is dedicated.

Reply:
Ok, Let's follow up the exist style currently.  Maybe we can define one similar common macro later to process such checks.